x/attribute/client/cli/cli_test.go

```diff
@@ -591,6 +593,83 @@ func (s *IntegrationTestSuite) TestAttributeTxCommands() {
 	}
 }
 
+func (s *IntegrationTestSuite) TestAttributeTxCommands_WithExpiration() {
+
+	testCases := []struct {
+		name            string
+		cmd             *cobra.Command
+		args            []string
+		expectErr       bool
+		respType        proto.Message
+		expectedCode    uint32
+		expectExpRecord bool
+	}{
+		{
+			"bind a new attribute name for testing",
+			namecli.GetBindNameCmd(),
+			[]string{
+				"txtestexp",
+				s.testnet.Validators[0].Address.String(),
+				"attribute",
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, s.testnet.Validators[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			false, &sdk.TxResponse{}, 0, false,
+		},
+		{
+			"set attribute with expiration",
+			cli.NewAddAccountAttributeCmd(),
+			[]string{
+				"txtestexp.attribute",
+				s.testnet.Validators[0].Address.String(),
+				"string",
+				"test value",
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, s.testnet.Validators[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+				fmt.Sprintf("--%s=%s", cli.FlagExpires, "1y"),
+			},
+			false, &sdk.TxResponse{}, 0, false,
+		},
+		{
+			"should find expiration record for attribute",
+			expcli.GetExpirationCmd(),
+			[]string{s.testnet.Validators[0].Address.String(),
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			false, &exptypes.QueryExpirationResponse{}, 0, true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			clientCtx := s.testnet.Validators[0].ClientCtx
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, tc.cmd, tc.args)
+
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), tc.respType), out.String())
+				switch t := tc.respType.(type) {
+				case *sdk.TxResponse:
+					s.Require().Equal(tc.expectedCode, t.Code)
```

Comment:
```suggestion
					s.Require().Equal(int(tc.expectedCode), int(t.Code), "response code from:\n%s", t)
```

By converting them to ints, they're easier to understand when they fail. Without that conversion, they are printed as hexadecimal, e.g. "Expected: 0x00, Actual: 0x10". That wouldn't be error code 10, it'd be 16. It's easy to confuse and waste time.

Also, by including the response in the message, it saves time trying to figure out what exactly the outcome was. This is crucial when tests fail intermittently, but still nice to have when it's a consistent failure.